api/plugins/action/deploy_target_config.py

```diff
@@ -90,35 +90,42 @@ def run(self, tmp=None, task_vars=None):
         return result
 
 def determine_required_updates(existing_configs, desired_configs):
-    addition_required = []  
-    deletion_required = [
-        config['id']
-        for config in existing_configs
-        if not any(
-            config['branches'] == desired['branches']
-            for desired in desired_configs
-        )
-    ]
-
+    addition_required = []
+    deletion_required = []
+
+    grouped_configs = {}
+    for config in existing_configs:
+        key = (config['branches'], config['pullrequests'], str(config['deployTarget']['id']), str(config['weight']))
+        if key not in grouped_configs:
+            grouped_configs[key] = []
+        grouped_configs[key].append(config)
+
+    #Identify duplicates and mark older ones for deletion
+    for key, configs in grouped_configs.items():
+        if len(configs) > 1:
+            sorted_configs = sorted(configs, key=lambda x: x['id'], reverse=True)
+            newest_config = sorted_configs[0]
+            for config in sorted_configs[1:]:
+                deletion_required.append(config['id'])
+            grouped_configs[key] = [newest_config]
+
+    # Adjusted logic for handling additions 
     for desired in desired_configs:
-        found = False
-        uptodate = True
-        for existing_config in existing_configs:
-            if existing_config['branches'] != desired['branches']:
-                continue
-
-            desired['_existing_id'] = existing_config['id']
-            found = True
-
-            # Mark for update (or in this context, addition) if there are discrepancies in any key property
-            if (existing_config['pullrequests'] != desired['pullrequests'] or
-                    str(existing_config['deployTarget']['id']) != str(desired['deployTarget']) or
-                    str(existing_config['weight']) != str(desired['weight'])):
-                desired['_existing_id'] = existing_config['id']
-                uptodate = False
-                break
-
-        if not found or not uptodate:
+        key = (desired['branches'], desired['pullrequests'], str(desired['deployTarget']), str(desired['weight']))
+        if key not in grouped_configs:
```

Comment:
As above, I think we could simplify this logic loop — you can modify `config` when you see it first and add a `exists` property rather than duplicating the `key` logic which will make this simpler to maintain going forward.